app/com/linkedin/drelephant/mapreduce/fetchers/MapReduceFetcherHadoop2.java

```diff
@@ -465,9 +465,20 @@ public static Matcher getDiagnosticMatcher(String diagnosticInfo) {
     return _LOCAL_DIAGNOSTIC_PATTERN.get().matcher(diagnosticInfo);
   }
 
-  public static JsonNode readJsonNode(URL url) throws IOException, AuthenticationException {
-    HttpURLConnection conn = _LOCAL_AUTH_URL.get().openConnection(url, _LOCAL_AUTH_TOKEN.get());
-    return _LOCAL_MAPPER.get().readTree(conn.getInputStream());
+  public static JsonNode readJsonNode(URL url, CloseableHttpClient httpClient) throws IOException, AuthenticationException {
+
+    HttpGet httpGet = new HttpGet(url.toString());
```

Comment:
u should handle exceptions here and close input stream in final block as long as InputStream is not null.

Reply:
@aryaKetan Addressed review comment. Please review.